storage/layers.go

```diff
@@ -624,14 +624,21 @@ func (r *layerStore) Changes(from, to string) ([]archive.Change, error) {
 	if !ok {
 		return nil, ErrLayerUnknown
 	}
-	fromLayer, ok := r.lookup(from)
-	if !ok {
-		fromLayer, ok = r.lookup(toLayer.Parent)
+	to = toLayer.ID
```

Comment:
It looks like this same block of code is copied three times, shouldn't we just add a new function to handle it?